src/python/pants/backend/awslambda/python/awslambda_python_rules.py

```diff
@@ -39,16 +41,32 @@
 
 @dataclass(frozen=True)
 class PythonAwsLambdaConfiguration(AWSLambdaConfiguration):
-    required_fields = (PythonAwsLambdaHandler,)
+    required_fields = (PythonAwsLambdaHandler, PythonAwsLambdaRuntime)
 
     handler: PythonAwsLambdaHandler
+    runtime: PythonAwsLambdaRuntime
 
 
 @dataclass(frozen=True)
 class LambdexSetup:
     requirements_pex: Pex
 
 
+class InvalidAWSLambdaRuntime(Exception):
+    pass
+
+
+def get_interpreter_from_runtime(runtime: str, addr: str) -> Tuple[int, int]:
+    """Returns the Python version implied by the runtime, as (major, minor)."""
+    mo = re.match(r"python(?P<major>\d)\.(?P<minor>\d+)", runtime)
+    if not mo:
+        raise InvalidAWSLambdaRuntime(
+            f"runtime field in python_awslambda target at {addr} must "
+            f"be of the form pythonX.Y, but was {runtime}"
+        )
+    return (int(mo["major"]), int(mo["minor"]))
```

Comment:
This logic better belongs in the `Field` definition, which will allow the Field to be reused in other places like if someone has an alternative implementation of AWS Lambda.

1) Move the validation into `compute_value`, which will make this validation eager when the value is first parsed (eager validation is good, I argue):

```python
PYTHON_INTERPRETER_REGEX = r"python(?P<major>\d)\.(?P<minor>\d+)"

@classmethod
def compute_value(cls, raw_value: Optional[str], *, address: Address) -> str:
   value = cast(str, super().compute_value(raw_value, address=address)
   if not re.match(cls.PYTHON_INTERPRETER_REGEX, value):
      raise InvalidFieldException("")
   return value
```

2) Add a method:

```python
def convert_to_interpreter(self) -> Tuple[int, int]:
   mo = re.match(cls.PYTHON_INTERPRETER_REGEX, self.value)
   return int(mo["major"]), int(mo["minor"])
```

Then, this rule will say `config.runtime.convert_to_interpreter()`.

Reply:
Actually, see below recommendation for something much simpler.

Reply:
Neat! Done.